WordPress/src/main/java/org/wordpress/android/ui/sitecreation/verticals/SiteCreationIntentsViewModel.kt

```diff
@@ -129,11 +129,17 @@ class SiteCreationIntentsViewModel @Inject constructor(
     }
 
     fun onSearchTextChanged(query: String) {
-        val searchResults = searchResultsProvider.search(fullItemsList.items, query)
+        val searchResults = searchResultsProvider.search(fullItemsList.items, query).toMutableList().apply {
+            if (query.isNotEmpty() && !(this.size == 1 && this[0].verticalText.equals(query, true))) {
```

Comment:
np: We could improve this by capturing the condition into a variable named something like `isFirstResultExactMatch`

Aside from this I really like the solution, neat adaptation to satisfy the new specs!!! 🙌

Reply:
I agree. I think we can ship this as is, and adjust this with a followup.

Reply:
Fixed in https://github.com/wordpress-mobile/WordPress-Android/pull/16242/commits/223202a9ac2095a3dd118342a54f23249e04b29a